src/pyqasm/visitor.py

```diff
@@ -1939,9 +1939,63 @@ def _visit_function_call(
 
         return return_value, result
 
-    def _visit_while_loop(self, statement: qasm3_ast.WhileLoop) -> None:
+    # Custom exceptions for break/continue
+    class BreakException(Exception):
         pass
 
+    class ContinueException(Exception):
+        pass
+
+    def _visit_break_statement(self, statement):
+        """Visit a break statement element."""
+        raise self.BreakException()
+
+    def _visit_continue_statement(self, statement):
+        """Visit a continue statement element."""
+        raise self.ContinueException()
+
+    def _visit_while_loop(self, statement: qasm3_ast.WhileLoop) -> list[qasm3_ast.Statement]:
+        """Visit a while loop statement element."""
+        result = []
+        loop_limit = getattr(self, "_loop_limit", 1000)
```

Comment:
```suggestion
        loop_limit = getattr(self, "_loop_limit", DEAFULT_MAX_LOOP_ITERATIONS)
```

Also, we'll need to add this attribute to the [visitor class during initialization](https://github.com/qBraid/pyqasm/blob/7076d46e057085fbb6189dbe52d6e6ab4b2b2170/src/pyqasm/visitor.py#L65) and passed in via the module